backend/app/services/knowledge/knowledge_service.py

```
@@ -128,69 +128,110 @@ def create_knowledge_base(
 
         # Check duplicate by Kind.name (unique identifier)
         existing_by_name = (
             db.query(Kind)
             .filter(
                 Kind.kind == "KnowledgeBase",
                 Kind.user_id == user_id,
-                Kind.namespace == data.namespace,
-                Kind.name == kb_name,
+                Kind.namespace == effective_namespace,
+                Kind.name == kind_name,
                 Kind.is_active == True,
```

Ruff (0.15.6)

[error] Avoid equality comparisons to `True`; use `Kind.is_active:` for truth checks

Replace with `Kind.is_active`

(E712)